app/cdap/components/PipelineDetails/PipelineDetailsTopPanel/PipelineDetailsButtons/PipelineHistoryButton.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Button } from '@material-ui/core';
+import React, { useEffect, useRef, useState } from 'react';
+import styled from 'styled-components';
+import classnames from 'classnames';
+import PipelineHistoryOuter from 'components/PipelineHistory/PipelineHistory';
+import HistoryIcon from '@material-ui/icons/History';
+import T from 'i18n-react';
+import { ApolloClient, gql, InMemoryCache } from 'apollo-boost';
+import { getCurrentNamespace } from 'services/NamespaceStore';
+import { useQuery } from '@apollo/react-hooks';
+import PaginationStepper from 'components/shared/PaginationStepper';
+
+const PREFIX = 'features.PipelineDetails.TopPanel';
+
+interface IPipelineHistoryButtonProps {
+  pipelineName: string;
+}
+
+export const PipelineHistoryButton = ({ pipelineName }: IPipelineHistoryButtonProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const buttonRef = useRef(null);
+
+  const toggleButton = () => {
+    setIsOpen(!isOpen);
+  };
+
+  const renderPipelineHistoryBtn = () => {
+    return (
+      <div onClick={toggleButton} className="btn pipeline-action-btn" ref={buttonRef} role="button">
```

Comment:
Please use a Mui Button for the button - if possible one in `shared/Buttons`. We don't want to use non-buttons as buttons because it makes a11y harder.